src/node.rs

```diff
@@ -11,21 +11,17 @@ impl<'a, T: Allocator> Node<'a, T> {
         Node { allocator, node }
     }
 
-    pub fn new_atom(&self, v: &[u8]) -> Self {
-        self.with_node(self.allocator.new_atom(v))
-    }
-
     pub fn with_node(&self, node: T::Ptr) -> Self {
         Node::new(self.allocator, node)
     }
 
-    pub fn sexp(&self) -> SExp<T::Ptr> {
+    pub fn sexp(&self) -> SExp<T::Ptr, T::AtomBuf> {
```

Comment:
Another possible idea would be to tweak `Node` so it takes `mut Allocator` instead of `&'a Allocator`, so it essentially "contains" the allocator. You could only have one `Node` alive at a time, but it would essentially "store" the allocator.

Reply:
the difficutly with that is that you couldn't have `first()`, `rest()` or the ability to iterate over it, because it couldn't create more `Node` objects

Reply:
I guess that doesn't really work because the `Result` doesn't return a `Node` in the error case, so the allocator would just be lost. Maybe it doesn't matter in that case since no more allocator references are done? Not sure. I suppose `Node` should really be renamed `Cursor` or `ProgramCounter` :) if we try this.

Reply:
I think those are the main reasons for `Node`'s existence in the first place

Reply:
I seem to think that my motivation was to provide a bunch of utility routines. A trick I didn't do was make it `Deref` to `Allocator`, which means we could call all the allocator methods on it like `new_atom`.

Reply:
Yeah, `first` and `rest` would have to consume the old node (so it could steal the allocator) or just return a `Ptr`. Not sure how worth it it would be. In any case, I really doubt it would make the code faster, and I'm not even sure if it would make it any nicer.

Reply:
It would get rid of the `'a` parameterization though.